feat: UIG-2739 - testen migreren naar componenten testen

code4cake opened the pull request:
Testen voor:

- [x] accessibility
   - [x] stories
   - [x] component
   - [x] child/
        - [x] compliance-status
        - [x] content.section
        - [x] header.section
        - [x] inaccessible-content.section
        - [x] setup-statement.section
        - [x] side-navigation.section
        - [x] title.section
        - [x] wcag-link
- [x] cookie-consent
    - [x] stories
    - [x] component
    - [x] child/
        - [x] consent-opt-in
        - [x] consent
- [x] cookie-statement
    - [x] stories
    - [x] component
    - [x] child/
        - [x] header
    - [x] cookie/
        - [x] vl-authentication-cookie
        - [x] vl-cookie
        - [x] vl-header-authentication
        - [x] vl-header-cookie
        - [x] vl-jsessionid-cookie
        - [x] vl-sticky-session-cookie
- [x] footer
    - [x] stories
    - [x] component
- [x] header
    - [x] stories
    - [x] component
- [x] privacy
    - [x] stories
    - [x] component
    - [x] child/
        - [x] header


---

[Bambo](https://www.milieuinfo.be/bamboo/browse/UIGOV-CUWC210)
[Jira - UIG-2739](https://www.milieuinfo.be/jira/browse/UIG-2739) - hoofd ticket commit `0d5ef91`

 Gedaan in dit ticket 

- [UIG-2739](https://www.milieuinfo.be/jira/browse/UIG-2739) - testen voor sections/* - commit `e8f5fbc` 
- [UIG-2730](https://www.milieuinfo.be/jira/browse/UIG-2730)  - imports cookie consent - **niks te comitten alle de imports waren al gefixed**
- [UIG-2740](https://www.milieuinfo.be/jira/browse/UIG-2740) - imports accessibility - commit `510765`

Goldflow commented on libs/sections/src/cookie-statement/child/header.section.ts:
```diff
@@ -1,4 +1,14 @@
+
+export const cookieStatementHeaderElements = (): WebComponent[] => [VlFunctionalHeaderComponent];
+
+// NOTE: Dit is de zelfde component van
```
maak hier gerust een nieuw ticket van in JIRA waarin je dit documenteert

code4cake replied:
gedaan - ticket gecreerd https://www.milieuinfo.be/jira/browse/UIG-2789

Goldflow reviewed (dismissed):
👍🏼

ksvlaanderen commented on apps/storybook-e2e/src/e2e/sections/accessibility/vl-accessibility.stories.cy.ts:
```diff
@@ -1,142 +1,13 @@
-            .shadow()
-            .find('h3')
-            .contains('De inhoud valt buiten de werkingssfeer van de toepasselijke wetgeving');
+        cy.get('vl-accessibility');
```
Kan je hier checken of er een shadow dom is? Deze test gaat altijd slagen ook al wordt de vl-accessibility niet gerenderd.

ksvlaanderen commented on apps/storybook-e2e/src/e2e/sections/cookie-consent/vl-cookie-consent.stories.cy.ts:
```diff
@@ -1,38 +1,16 @@
-        cy.get('script#vl-cookie-consent-matomo-script')
-            .should('contain.text', `_paq.push(['setSiteId', ${matomoId}]);`)
-            .should('contain.text', `var u='${matomoUrl}'`);
+        cy.get('vl-cookie-consent');
```
Shadow DOM afchecken.

ksvlaanderen commented on apps/storybook-e2e/src/e2e/sections/cookie-consent/vl-cookie-consent.stories.cy.ts:
```diff
@@ -1,38 +1,16 @@
     });
 
-    it('should emit event on close', () => {
+    it('should contain the `Cookie-toestemming`', () => {
```
Deze test kan ook geplaatst worden bij de component tests van dit component.

code4cake replied:
het bestaat ook in de component testen `vl-cookie-consent.section.cy.ts`,  maar in de storybook om de `cookie consent` echt te krijgen moet ik de btn in de story op klikken. Dus ik test gewoon dat als je de btn op klikt zie je de story, maar alles qua functionaliteit is getest binnen `vl-cookie-consent.section.cy.ts`

ksvlaanderen commented on libs/common/utilities/src/util/utils.ts:
```diff
@@ -9,9 +10,9 @@ declare const vl: VL;
+ * @param webComponents {WebComponent[]} - een array van web-components
  */
-export const registerWebComponents = (webComponents: any[]) => {};
+export const registerWebComponents = (webComponents: WebComponent[]) => {};
```
Klopt dit? Hier zeg je dat `webComponents` een array van arrays kan bevatten.

ksvlaanderen reviewed (changes requested):
Een aantal dingen die doorgetrokken moeten worden voor alle testen:

- Bij een story test altijd de shadow DOM afchecken, als je enkel het bestaan van de custom tag in de DOM afchecked gaat de test altijd slagen ook al rendert de component niet
- De naam van een describe altijd laten beginnen met het type van de test, e.g. 'story vl-accessibility - ...' of 'component vl-accessibility - ...'

kspeltix reviewed (commented):
- ik zie dat in deze PR 3 tickets worden opgelost: UIG-2739 / UIG-2730 / UIG-2740
  -> je mag gerust meerdere tickets in 1 branch oplossen (als ze aan elkaar gerelateerd zijn)
  -> maar het is wel de bedoeling om een commit per ticket te behouden, dus in dat geval NIET squashen tot 1 commit
  -> kan je dit ticket dus terug opsplitsen in 3 commits ?
- de wijziging in cypress/support/component.ts is die nodig ?
  -> zo ja kan je dan een lijntje commentaar toevoegen waarom deze implementatie nodig is
   -> en de vorige implementatie dan ook verwijderen en niet gewoon in commentaar zetten
- cypress/support/component-index.html
  -> meta tags moeten niet gesloten worden, zie https://html.com/tags/meta/
  -> gelieve niet zomaar zaken te wijzigen die niet fout zijn
- common/utilities
  -> je maakt een WebComponent alias van Class
  -> aliassen zijn altijd verwarrend want dan heb je 2 namen voor 1 ding
  -> het klopt ook niet in dit geval want er is niet zoiets als een basis klasse voor een WebComponent
  -> een basis klasse van een WebComponent zou een combinatie zijn van BaseLitElement en het  BaseElementOfType gebeuren (maar dat is een probleem op zich)
  -> ooit gaan al onze componenten extenden van BaseLitElement, maar in afwachting van dat zou ik het niet complexer maken en daar waar je nu WebComponent of WebComponent[] wil gebruiken gewoon 'any' of 'any[]' gebruiken (zo doen we het nu ook)
  -> dus gelieve alle wijzigingen in common/utilities te reverten en de code aan te passen zodat het WebComponent type niet meer nodig is

Goldflow commented on libs/sections/src/cookie-statement/vl-cookie-statement.section.cy.ts:
```diff
@@ -0,0 +1,119 @@
+    disableBackLink: false,
+};
+
+const vlCookieStatementElement = () => cy.get('vl-cookie-statement');
```
normaal gezien maken we geen abstractie voor zoiets; het bespaart ook niet per sé lijnen code uit

Goldflow commented on libs/sections/src/cookie-statement/vl-cookie-statement.section.cy.ts:
```diff
@@ -0,0 +1,119 @@
+    });
+
+    it('should mount', () => {
+        vlCookieStatementElement();
```
als er een shadow dom is, die ook aftesten bij de `mount` test

Goldflow commented on libs/sections/src/cookie-statement/cookie/vl-cookie.section.ts:
```diff
@@ -1,9 +1,39 @@
+        ]);
+    }
+
+    // constructor({ title, name, purpose, domain, processor, validity }: any = {}) {
```
mag dit commentaar weg?

Goldflow commented on apps/storybook-e2e/src/e2e/sections/privacy/vl-privacy.stories.cy.ts:
```diff
@@ -2,68 +2,10 @@ const privacyUrl = 'http://localhost:8080/iframe.html?id=sections-privacy--priva
-        cy.get('vl-privacy').shadow().find('h2#privacy-declaration').scrollIntoView();
-
-        shouldHaveExpandedToggle('#privacy-declaration', true);
+        cy.get('vl-privacy');
```
`.shadow()` toevoegen indien component een shadow dom heeft

Goldflow commented on libs/sections/src/footer/vl-footer.section.cy.ts:
```diff
@@ -0,0 +1,80 @@
+    `);
+};
+
+const vlFooterElement = () => cy.get('vl-footer');
```
we maken normaal gezien geen abstractie voor zoiets eenvoudig; dit bespaart ook geen code uit

Goldflow commented on libs/sections/src/header/vl-header.section.cy.ts:
```diff
@@ -1,30 +1,168 @@
-        beforeEach(() => {
-            mountHeaderWithSkeleton();
-        });
+const vlHeaderElement = () => cy.get('vl-header');
```
we maken normaal gezien geen abstractie voor zoiets maar schrijven het dan voluit

Goldflow commented on libs/sections/src/privacy/vl-privacy.section.cy.ts:
```diff
@@ -0,0 +1,128 @@
+        ${unsafeHTML(props.headerSlot)}
+    </vl-privacy>`);
+
+const vlPrivacyElement = () => cy.get('vl-privacy');
```
normaal gezien maken we hier geen abstractie voor

Goldflow commented on libs/sections/src/privacy/vl-privacy.section.cy.ts:
```diff
@@ -0,0 +1,128 @@
+    });
+
+    it('should mount', () => {
+        vlPrivacyElement();
```
ook shadow dom aftesten `.shadow();`

Goldflow committed:
feat: UIG-2740 - vl-accessibility, vl-cookie-consent - import problemen opgelost

De wijzigingen bevatten ook UIG-2730, waarvan de inhoud overlapt met dit ticket.

Goldflow committed:
feat: UIG-2739 - sections testen gemigreerd

Storybook testen gemigreerd naar component testen & testen verbeterd.

kspeltix commented:
:tada: This PR is included in version 1.25.0 :tada:

The release is available on [GitHub release](https://github.com/milieuinfo/uigov-web-components/releases/tag/v1.25.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
